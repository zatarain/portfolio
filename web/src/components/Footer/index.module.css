.footer {
	font-size: 0.85em;
	background-color: var(--bg-secondary);
	color: var(--fg-secondary);
	text-align: left;
	box-shadow: 0 -5px 15px 0 var(--bg-secondary);
	margin-top: 3rem;
	padding: 1rem;
}

.footer h4 {
	margin: 0;
}

.footer ul {
	list-style: none;
	padding: 0;
	margin: 0;
	line-height: 1.5em;
}

.footer>ul {
	display: flex;
	justify-content: space-evenly;
	justify-items: center;
	flex-wrap: wrap;
	gap: 1rem;
}

.footer a:link,
.footer a:visited {
	color: var(--link-secondary);
}

.footer a:hover,
.footer a:active {
	color: var(--highlight-secondary);
}

@media screen and (max-width: 820px) {
	.footer>ul>li {
		flex-basis: 14rem;
	}
}

@media screen and (max-width: 700px) {
	.footer>ul>li {
		flex-basis: 10rem;
	}
}
