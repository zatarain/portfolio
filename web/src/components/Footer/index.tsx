import Link from 'next/link'
import styles from './index.module.css'

const Footer = () => {
	return (
		<footer className={styles.footer}>
			<ul>
				<li>
					<h4>Tech Stuff</h4>
					<ul>
						<li><Link href="#">C++</Link></li>
						<li><Link href="#">TypeScript</Link></li>
						<li><Link href="#">Ruby</Link></li>
						<li><Link href="#">Terraform</Link></li>
						<li><Link href="#">Golang</Link></li>
						<li><Link href="#">Python</Link></li>
						<li><Link href="#">Cloud - AWS</Link></li>
						<li><Link href="#">Git</Link></li>
					</ul>
				</li>
				<li>
					<h4>Skills</h4>
					<ul>
						<li><Link href="#">Algorithms and Data Structures</Link></li>
						<li><Link href="#">Mathematics</Link></li>
						<li><Link href="#">Software Engineering</Link></li>
						<li><Link href="#">Web Development</Link></li>
						<li><Link href="#">Continuous Integration</Link></li>
						<li><Link href="#">Tech Agnostic</Link></li>
						<li><Link href="#">Proactivity</Link></li>
						<li><Link href="#">Continuous Learning</Link></li>
					</ul>
				</li>
				<li>
					<h4>Personal Projects</h4>
					<ul>
						<li><Link href="#">Algorists Group</Link></li>
						<li><Link href="#">Unit Test Zeal</Link></li>
						<li><Link href="#">Personal Website</Link></li>
						<li><Link href="#">Jarnik Build System</Link></li>
						<li><Link href="#">Spectral Clustering</Link></li>
					</ul>
				</li>
				<li>
					<h4>Volunteering</h4>
					<ul>
						<li><Link href="#">LEGO Robotics and Tech for Kids</Link></li>
						<li><Link href="#">Plant Trees</Link></li>
						<li><Link href="#">OmegaUp Latinoamerica</Link></li>
						<li><Link href="#">Rider Photoshoot</Link></li>
						<li><Link href="#">Charity Events</Link></li>
					</ul>
				</li>
				<li>
					<h4>Hobbies and fun</h4>
					<ul>
						<li><Link href="#">Cooking</Link></li>
						<li><Link href="#">Liverpool FC</Link></li>
						<li><Link href="#">Anime</Link></li>
						<li><Link href="#">Videogames</Link></li>
						<li><Link href="#">Boardgames</Link></li>
						<li><Link href="#">Watching sports</Link></li>
						<li><Link href="#">Travel and Hiking</Link></li>
					</ul>
				</li>
				<li>
					<h4>Hates and boring</h4>
					<ul>
						<li><Link href="#">Apple stuff</Link></li>
						<li><Link href="#">Real Madrid FC</Link></li>
						<li><Link href="#">Java stuff</Link></li>
					</ul>
				</li>
			</ul>
		</footer>
	)
}

export default Footer
