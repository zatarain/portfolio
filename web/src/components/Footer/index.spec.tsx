import { render } from '@testing-library/react'
import Footer from '.'

describe('<Footer />', () => {
	it('renders the component correctly', () => {
		const { asFragment } = render(<Footer />)

		expect(asFragment()).toMatchSnapshot()
	})
})
