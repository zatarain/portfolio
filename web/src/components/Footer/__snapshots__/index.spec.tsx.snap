// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the component correctly 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <ul>
      <li>
        <h4>
          Tech Stuff
        </h4>
        <ul>
          <li>
            <a
              href="#"
            >
              C++
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              TypeScript
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Ruby
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Terraform
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Golang
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Python
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Cloud - AWS
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Git
            </a>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Skills
        </h4>
        <ul>
          <li>
            <a
              href="#"
            >
              Algorithms and Data Structures
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Mathematics
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Software Engineering
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Web Development
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Continuous Integration
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Tech Agnostic
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Proactivity
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Continuous Learning
            </a>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Personal Projects
        </h4>
        <ul>
          <li>
            <a
              href="#"
            >
              Algorists Group
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Unit Test Zeal
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Personal Website
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Jarnik Build System
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Spectral Clustering
            </a>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Volunteering
        </h4>
        <ul>
          <li>
            <a
              href="#"
            >
              LEGO Robotics and Tech for Kids
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Plant Trees
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              OmegaUp Latinoamerica
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Rider Photoshoot
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Charity Events
            </a>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Hobbies and fun
        </h4>
        <ul>
          <li>
            <a
              href="#"
            >
              Cooking
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Liverpool FC
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Anime
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Videogames
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Boardgames
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Watching sports
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Travel and Hiking
            </a>
          </li>
        </ul>
      </li>
      <li>
        <h4>
          Hates and boring
        </h4>
        <ul>
          <li>
            <a
              href="#"
            >
              Apple stuff
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Real Madrid FC
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Java stuff
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
